src/xdp/program.c

```diff
@@ -125,6 +160,89 @@ XdpInitializeFrameCache(
     Cache->Flags = 0;
 }
 
+static
+_Must_inspect_result_
+ebpf_result_t
+EbpfExtensionInvokeProgram(
+    _In_ const EBPF_EXTENSION_CLIENT *Client,
+    _In_ const VOID *Context,
+    _Out_ uint32_t *Result)
+{
+    ebpf_invoke_program_function_t InvokeProgram = Client->InvokeProgram;
+    const void* ClientBindingContext = Client->ClientBindingContext;
+
+    return InvokeProgram(ClientBindingContext, Context, Result);
+}
+
+static
+XDP_RX_ACTION
+XdpInvokeEbpf(
+    _In_ HANDLE EbpfTarget,
+    _In_ XDP_FRAME *Frame,
+    _In_opt_ XDP_RING *FragmentRing,
+    _In_opt_ XDP_EXTENSION *FragmentExtension,
+    _In_ UINT32 FragmentIndex,
+    _In_ XDP_EXTENSION *VirtualAddressExtension
+    )
+{
+    const EBPF_EXTENSION_CLIENT *Client = (const EBPF_EXTENSION_CLIENT *)EbpfTarget;
+    XDP_BUFFER *Buffer;
+    UCHAR *Va;
+    EBPF_XDP_MD XdpMd;
+    ebpf_result_t EbpfResult;
+    XDP_RX_ACTION RxAction;
+    UINT32 Result;
+
+    UNREFERENCED_PARAMETER(FragmentIndex);
+
+    ASSERT((FragmentRing == NULL) || (FragmentExtension != NULL));
+
+    //
+    // Fragmented frames are currently not supported by eBPF.
```

Comment:
If this is a gap in the ebpf-for-windows repo, is there an issue filed?  Or is this just a TODO to be handled later in this draft PR?

Reply:
This will not be handled later in this PR - an issue will be filed.

Reply:
Please link the issue in the comment for later follow up.